src/components/carousel/index.tsx

```diff
@@ -437,17 +441,17 @@ class Carousel extends Component<CarouselProps, CarouselState> {
       : undefined;
     const snapToOffsets = this.getSnapToOffsets();
     const marginBottom = Math.max(0, this.getContainerPaddingVertical() - 16);
-
     return (
       <View animated={animated} style={[{marginBottom}, containerStyle]} onLayout={this.onContainerLayout}>
         <ScrollView
           {...others}
           ref={this.carousel}
           contentContainerStyle={scrollContainerStyle}
-          horizontal
+          horizontal={this.props.horizontal}
```

Comment:
We usually like to destruct props at the start of a method. 
I suggest moving `horizontal ` prop together with the others.. (: